File: Entity Framework/Kod/04/Invoice.cs
using System.ComponentModel.DataAnnotations;

namespace ŁukaszDydekEFProducts
{
    internal class Invoice
    {
        [Key]
        public int InvoiceNumber { get; set; }
        public int Quantity { get; set; } = 0;
        public List<ProductInvoice> ProductInvoices { get; } = new();
        public void AddProduct(ProductInvoice productInvoice, int quantity)
        {
            ProductInvoices.Add(productInvoice);   
            this.Quantity += quantity;
        }
    }
}

File: Hibernate & JPA/Kod/08/Faktury wraz z produktami/Invoice.java
package org.example;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "Invoices")
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int invoiceNumber;
    private int quantity = 0;

    @ManyToMany(cascade = CascadeType.ALL, mappedBy = "invoices")
    private List<Product> products = new ArrayList<>();

    public Invoice() {}

    public void addProduct(Product product, int quantity) {
        this.products.add(product);
        this.quantity += quantity;
    }
}
